torch/nn/quantized/modules/embedding.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch
+import torch.nn as nn
+from torch import Tensor  # noqa: F401
+from torch._jit_internal import Optional, List  # noqa: F401
+from torch.nn.quantized.modules.utils import hide_packed_params_repr
+from torch.nn.quantized.modules.utils import _quantize_weight
+
+class EmbeddingPackedParams(torch.nn.Module):
+    _version = 1
+
+    def __init__(self, num_embeddings, embedding_dim, dtype=torch.quint8):
+        super(EmbeddingPackedParams, self).__init__()
+        self.dtype = dtype
+        if self.dtype == torch.quint8:
+            scales = torch.ones(num_embeddings, dtype=torch.float)
+            zero_points = torch.ones(num_embeddings, dtype=torch.float)
+            wq = torch._empty_per_channel_affine_quantized([num_embeddings, embedding_dim], scales=scales,
+                                                           zero_points=zero_points,
+                                                           axis=0, dtype=torch.quint8)
+            self.set_weight(wq)
+        else:
+            raise RuntimeError('Unsupported dtype on quantized embedding!')
+
+    @torch.jit.export
+    def set_weight(self, weight):
+        # type: (torch.Tensor) -> None
+        if self.dtype == torch.quint8:
+            self._packed_weight = torch.ops.quantized.embedding_bag_prepack(weight)
+        else:
+            raise RuntimeError('Unsupported dtype on quantized embedding!')
+
+
+    @torch.jit.export
+    def _weight(self):
+        if self.dtype == torch.quint8:
+            return torch.ops.quantized.embedding_bag_unpack(self._packed_weight)
+        else:
+            raise RuntimeError('Unsupported dtype on quantized embedding!')
+
+    def forward(self, x):
+        return x
+
+    # Version 1
+    #   self
+    #   |--- _packed_weight : Tensor representing weight of EmbeddingPackedParamsBase
+    #   |--- dtype : torch.dtype
+
+    def _save_to_state_dict(self, destination, prefix, keep_vars):
```

Comment:
Should we also have a field for bitwidth?

Reply:
we can use the tensor dtype to determine the bitwidth, right? Currently it only supports 8bit, but once we add 4-bit  qtensor it should be encoded in the dtype.